src/inference/Inference.jl

```diff
@@ -259,86 +259,38 @@ end
 ##########################
 
 function _params_to_array(ts::Vector{T}, spl::Sampler) where {T<:AbstractTransition}
-    names = Set{String}()
-    dicts = Vector{Dict{String, Any}}()
-
+    names_set = Set{String}()
     # Extract the parameter names and values from each transition.
-    for t in ts
+    dicts = map(enumerate(ts)) do (i, t)
         nms, vs = flatten_namedtuple(t.θ)
-        push!(names, nms...)
-
-        # Convert the names and values to a single dictionary.
-        d = Dict{String, Any}()
-        for (k, v) in zip(nms, vs)
-            d[k] = v
-        end
-        push!(dicts, d)
-    end
-    
-    # Convert the set to an ordered vector so the parameter ordering 
-    # is deterministic.
-    ordered_names = collect(names)
-    vals = Matrix{Union{Real, Missing}}(undef, length(ts), length(ordered_names))
-
-    # Place each element of all dicts into the returned value matrix.
-    for i in eachindex(dicts)
-        for (j, key) in enumerate(ordered_names)
-            vals[i,j] = get(dicts[i], key, missing)
+        foreach(nms) do nm
+            push!(names_set, nm)
         end
+        # Convert the names and values to a single dictionary.
+        return Dict(nms[j] => vs[j] for j in 1:length(vs))
     end
+    names = collect(names_set)
+    vals = [get(dicts[i], key, missing) for i in eachindex(dicts), 
+        (j, key) in enumerate(names)]
 
-    return ordered_names, vals
+    return names, vals
 end
 
 function flatten_namedtuple(nt::NamedTuple{pnames}) where {pnames}
-    vals = Vector{Real}()
-    names = Vector{AbstractString}()
-
-    for k in pnames
+    temp = map(keys(nt)) do k
```

Comment:
It seems you could just use `mapreduce`?

```suggestion
    names_vals = mapreduce(vcat, keys(nt)) do k
```